pkg/controller/volume/expand/expand_controller.go

```diff
@@ -123,107 +108,151 @@ func NewExpandController(
 	expc.recorder = eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "volume_expand"})
 	blkutil := volumepathhandler.NewBlockVolumePathHandler()
 
-	expc.opExecutor = operationexecutor.NewOperationExecutor(operationexecutor.NewOperationGenerator(
+	expc.operationGenerator = operationexecutor.NewOperationGenerator(
 		kubeClient,
 		&expc.volumePluginMgr,
 		expc.recorder,
 		false,
-		blkutil))
-
-	expc.resizeMap = cache.NewVolumeResizeMap(expc.kubeClient)
+		blkutil)
 
 	pvcInformer.Informer().AddEventHandler(kcache.ResourceEventHandlerFuncs{
-		UpdateFunc: expc.pvcUpdate,
-		DeleteFunc: expc.deletePVC,
+		AddFunc: expc.enqueuePVC,
+		UpdateFunc: func(old, new interface{}) {
+			oldPVC := old.(*v1.PersistentVolumeClaim)
+			oldSize := oldPVC.Spec.Resources.Requests[v1.ResourceStorage]
+
+			newPVC := new.(*v1.PersistentVolumeClaim)
+			newSize := newPVC.Spec.Resources.Requests[v1.ResourceStorage]
+			if newSize.Cmp(oldSize) > 0 {
+				expc.enqueuePVC(new)
+			}
+		},
+		DeleteFunc: expc.enqueuePVC,
```

Comment:
Do we need to forget the key PVC was deleted?

Reply:
It will be forgotten if it DNE in the informer